builtin/providers/cidr/provider_test.go

```diff
@@ -0,0 +1,48 @@
+package cidr
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/builtin/providers/aws"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+var testProviders = map[string]terraform.ResourceProvider{
+	"cidr": Provider(),
+}
+var testAccProviders map[string]terraform.ResourceProvider
+var testAccProvider *schema.Provider
+var testAccAwsProvider *schema.Provider
+
+func init() {
+	testAccProvider = Provider().(*schema.Provider)
+	testAccAwsProvider = aws.Provider().(*schema.Provider)
+	testAccProviders = map[string]terraform.ResourceProvider{
+		"cidr": testAccProvider,
+		"aws":  testAccAwsProvider,
```

Comment:
If at all possible I'd like to avoid depending on any other providers here -- particularly ones that require accounts and can incur costs -- since it makes it more complex and expensive to run the tests, and more likely to fail due to upstream changes, etc.

Do you think it would be reasonable to cut the AWS-specific parts out of the acceptance tests and just assert directly on the data source results? I know this is not as "deep" a test, but I think this will make it more likely that future contributors to this provider will be able to run the tests, and will also smooth the process of us bringing it into the official release process since we won't need to coordinate with giving it access to the AWS acceptance test environment.